src/web_interface/templates/user_management/edit_user.html

```diff
@@ -72,16 +36,53 @@ <h3 class="mb-3">Manage user {{ user.email }}</h3>
                     Roles
                 </th>
                 <td>
-                    <a href="#" id="roles" data-type="checklist" data-pk="{{ user.email }}" data-url="/admin/edit_user" data-title="Select User Groups"></a>
+                    <ul class="list-group">
+                        {% for role in user.roles %}
+                            <li class="list-group-item">{{ role.name | replace_underscore }}</li>
+                        {% endfor %}
+                    </ul>
+                    <button class="btn btn-primary btn-sm" data-toggle="modal" data-target="#change_user_roles_modal" style="width: 100%">
+                        Edit Roles
+                    </button>
                 </td>
             </tr>
         </table>
 
-        <!-- role help -->
-        <div id="user_role_help_modal" class="modal" role="dialog" data-backdrop="false">
+        <!-- change user roles modal -->
+        <div id="change_user_roles_modal" class="modal fade" role="dialog">
             <div class="modal-dialog">
                 <div class="modal-content">
+                    <div class="modal-header d-flex justify-content-between align-items-center">
+                        <h5 class="modal-title">Select User roles</h5>
+                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span></button>
+                    </div>
 
+                    <div class="modal-body">
+                        <form action="/admin/edit_user/{{ user_id }}" method="post">
+                            <div class="form-group">
+                                {% for role in available_roles %}
+                                    <div class="custom-control custom-switch" id="user_roles">
+                                        <input type="checkbox" class="custom-control-input" id="{{ role }}" value="{{ role }}" name="input_roles"
+                                               {% if role in user.roles %} checked {% endif %}>
+                                        <label class="custom-control-label" for="{{ role }}">{{ role | replace_underscore }}</label>
+                                    </div>
+                                {% endfor %}
+                            </div>
+                            <button class="btn btn-primary btn-sm" type="submit" style="width: 100%" id="input_roles" value=submit>
+                                Submit
+                            </button>
+                        </form>
+                    </div>
+
+                </div>
+            </div>
+        </div>
+
+
+        <!-- role help -->
+        <div id="user_role_help_modal" class="modal" role="dialog" data-backdrop="false">
+            <div class="modal-dialog modal-lg">
+                <div class="modal-content">
```

Comment:
On a side note: Is it wise to allow superusers to delete themselves? I just locked me out of the instance by doing so :smile:

Reply:
Probably not but who are we to disallow them to do it? :sweat_smile:

Reply:
Fair enough.